pyquil/quilbase.py

```diff
@@ -527,6 +530,33 @@ def format_parameter(element):
     assert False, "Invalid parameter: %r" % element
 
 
+def check_for_pi(element):
+    """
+    Check to see if there exists a rational number r = p/q
+    in reduced form for which the difference between element/np.pi
+    and r is small and q <= 8.
+
+    :param element: float
+    :return element: pretty print string if true, else standard representation.
+    """
+    frac = Fraction(element / np.pi).limit_denominator(8)
+    num, den = frac.numerator, frac.denominator
+    sign = "-" if num < 0 else ""
+    if np.isclose(num / float(den), element / np.pi):
+        if num == 0:
+            return "0"
+        elif abs(num) == 1 and not den == 1:
+            return sign + "pi/" + repr(den)
+        elif not abs(num) == 1 and den == 1:
+            return repr(num) + "*pi"
+        elif abs(num) == 1 and den == 1:
```

Comment:
These checks could be cleaned up a little bit more by putting this condition first:
if abs(num) == 1 and den == 1:
elif abs(num) == 1:
elif den == 1:
else:

Reply:
Thanks for the suggestion.  I also removed flake8 --ignore=E226 from tox.ini (reverted it back to the original) and it builds successfully now.  The problem was with white spaces between my division and multiplication operators.

Reply:
Yes, this does make things clearer.  Only different is I added:
```
if num == 0:
            return "0"
elif abs(num) == 1 and den == 1:
    etc.
```
to the beginning, so I can convert float(0) to int(0).